vowpalwabbit/active.cc

```diff
@@ -150,6 +150,7 @@ base_learner* active_setup(options_i& options, vw& all)
     return nullptr;
 
   data->all = &all;
+  data->m_random_state = &(all.random_state);
```

Comment:
If we made this a shared_ptr and added a get_xyz() to VW, the life time of this service (random number generation) is no longer tied to the lifetime of vw object.

Reply:
Eventually we can define a service provider interface around this and potentially have other ways of getting this service.

---
In reply to: [320818351](https://github.com/VowpalWabbit/vowpal_wabbit/pull/2069#discussion_r320818351) [](ancestors = 320818351)